src/Symfony/Component/Notifier/Bridge/Mercure/Tests/MercureTransportFactoryTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mercure\Tests;
+
+use LogicException;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Mercure\PublisherInterface;
+use Symfony\Component\Notifier\Bridge\Mercure\MercureTransportFactory;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+final class MercureTransportFactoryTest extends TestCase
```

Comment:
As Mercure has a special transport factory (because of the locator), I extended the abstract class but also added an extra test: `testNotFoundPublisherThrows`